### PR TITLE
net: move the _SF*/_SS* definitions to net.h

### PR DESCRIPTION
**Pull request by anchao**
Build break caused by 23b8b397999136c61b3c834956ad1da55fb0e072

Change-Id: I4a63ede90fbe3385423a9d88ae811cd596491fec
Signed-off-by: chao.an <anchao@xiaomi.com>

### CODE REVIEW

File: include/nuttx/net/net.h [hunk 1]
```diff
@@ -59,6 +59,32 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
+/* Definitions of 8-bit socket flags */
```
Comment (xiaoxiang781216) [inline include/nuttx/net/net.h]: Move after line 110, so all socket flag in one location

Comment (anchao) [reply include/nuttx/net/net.h]: Done

**Comment by patacongo** [comment]
Thanks!  I am very happy to have that resolved satisfactorily.  I think there is an issue with mixing network socket knowledge inside of the scheduler task start-up logic.  That is not a clean separation of functionality.  But neither is mixing knowledge of the the file descriptor structures.  So we are really no worse than before.

It is very important to keep the strict modularity of the OS internal design.  That is the only way that this OS can survive into the future without degenerating into an unmaintainable mass of spaghetti code.  We are not doing a very good job of enforcing the architecture.. I am not doing a good job either.  We need more people with the capability to review internal OS changes and with a good understanding of the modularity objectives.

**Comment by xiaoxiang781216** [comment]
Yes, this is also our target: the modularity is very important for NuttX.

**Pull request merged**